src/Microsoft.VisualStudio.ProjectSystem.Managed/MsBuildUtilities.cs

```diff
@@ -0,0 +1,172 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    /// <summary>
+    /// Utitlies class to manipulate MsBuild projects.
+    /// </summary>
+    internal abstract class MsBuildUtilities
+    {
+        /// <summary>
+        /// Gets a project property.
+        /// </summary>
+        /// <param name="project">Xml representation of the MsBuild project.</param>
+        /// <param name="propertyName">Name of the property.</param>
+        /// <returns>Requested project property. Null if the property is not present.</returns>
+        public static ProjectPropertyElement GetProperty(ProjectRootElement project, string propertyName)
+        {
+            Requires.NotNull(project, "project");
+
+            return project.Properties
+                .Where(p => string.Equals(p.Name, propertyName, StringComparison.OrdinalIgnoreCase))
+                .FirstOrDefault();
```

Comment:
For writing we need to use construction not evaluation because we want to write only to the project file. The way things are we will have the defaults on the SDK so if you wanted to remove and tried operating on the evaluated model the OM would be trying to remove that from the property defined in the SDK which isn't what you want. In all of these cases what we do is get the evaluated value, process it and then stamp the changed property on the project.

I know this is a weird mix so if that isn't clear lets chat so that I can understand better what's confusing and update things (be it documentation or code changes) accordingly.